src/AutoBuild.jl

```diff
@@ -305,11 +309,12 @@ function get_compilers_versions(; compilers = [:c])
     return output
 end
 
-function upload_to_github_releases(repo, tag, path; attempts::Int = 3, verbose::Bool = false)
+function upload_to_github_releases(repo, tag, path; gh_auth=github_auth(;allow_anonymous=false), 
+                                   attempts::Int = 3, verbose::Bool = false)
     for attempt in 1:attempts
         try
             ghr() do ghr_path
-                run(`$ghr_path -u $(dirname(repo)) -r $(basename(repo)) $(tag) $(path)`)
+                run(`$ghr_path -u $(dirname(repo)) -r $(basename(repo)) -t $(gh_auth.token) $(tag) $(path)`)
```

Comment:
You know, I'd always been providing this to BB via the environment variable and `ghr` was picking it up through those being inherited, but this is clearly superior.  Nice work.

Reply:
Thanks!